azurerm/internal/services/network/data_source_virtual_machine_scale_set_network_interfaces.go

```diff
@@ -0,0 +1,275 @@
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+
+						"enable_accelerated_networking": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"enable_ip_forwarding": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"private_ip_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"private_ip_addresses": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceArmVirtualMachineScaleSetNetworkInterfacesRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.InterfacesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resGroup := d.Get("resource_group_name").(string)
+	vmssName := d.Get("virtual_machine_scale_set_name").(string)
+
+	resp, err := client.ListVirtualMachineScaleSetNetworkInterfaces(ctx, resGroup, vmssName)
+	if err != nil {
+		return fmt.Errorf("Error making Read request on Azure VMSS %q (Resource Group %q): %+v", vmssName, resGroup, err)
+	}
+
+	results := make([]interface{}, 0)
+
+	for _, iface := range resp.Values() {
+		result := make(map[string]interface{})
+
+		if iface.Name != nil {
+			result["name"] = *iface.Name
+		} else {
+			result["name"] = ""
+		}
+
+		if iface.NetworkSecurityGroup != nil {
+			result["network_security_group_id"] = *iface.NetworkSecurityGroup.ID
+		} else {
+			result["network_security_group_id"] = ""
+		}
+
+		if iface.MacAddress != nil {
+			result["mac_address"] = *iface.MacAddress
+		} else {
+			result["mac_address"] = ""
+		}
+
+		if iface.VirtualMachine != nil {
+			result["virtual_machine_id"] = *iface.VirtualMachine.ID
+		} else {
+			result["virtual_machine_id"] = ""
+		}
+
+		if iface.ID != nil {
+			result["id"] = *iface.ID
+		} else {
+			result["id"] = ""
+		}
+
+		if iface.IPConfigurations != nil && len(*iface.IPConfigurations) > 0 {
+			configs := *iface.IPConfigurations
+
+			result["private_ip_address"] = *configs[0].InterfaceIPConfigurationPropertiesFormat.PrivateIPAddress
+
+			addresses := make([]interface{}, 0)
+			for _, config := range configs {
+				if config.InterfaceIPConfigurationPropertiesFormat != nil {
+					addresses = append(addresses, config.InterfaceIPConfigurationPropertiesFormat.PrivateIPAddress)
+				}
+			}
+			result["private_ip_addresses"] = addresses
+		}
+
+		if iface.IPConfigurations != nil {
+			result["ip_configuration"] = flattenNetworkInterfaceIPConfigurations(iface.IPConfigurations)
+		}
+
+		var appliedDNSServers []string
+		var dnsServers []string
+		if dnsSettings := iface.DNSSettings; dnsSettings != nil {
+			if s := dnsSettings.AppliedDNSServers; s != nil {
+				appliedDNSServers = *s
+			}
+
+			if s := dnsSettings.DNSServers; s != nil {
+				dnsServers = *s
+			}
```

Comment:
Actually we have a built-in helper function `utils.FlattenStringSlice()`, could we please use this function instead? By doing this, these nil-checks can be removed.